src/briefcase/commands/run.py

```diff
@@ -1,23 +1,217 @@
+import re
+import signal
 from abc import abstractmethod
 from typing import Optional
 
 from briefcase.config import BaseConfig
-from briefcase.exceptions import BriefcaseCommandError
+from briefcase.exceptions import BriefcaseCommandError, BriefcaseTestSuiteFailure
 
 from .base import BaseCommand, full_options
 
 
+class LogFilter:
+    DEFAULT_SUCCESS_REGEX = (
+        # Unittest
+        # The a{123} groups are to support Android logging differences.
+        # when https://github.com/chaquo/chaquopy/issues/746 is resolved,
+        # those groups can be deleted.
+        r"(^-{65,}\n(?P<a1> \n)?Ran \d+ tests in \d+\.\d{3}s\n(?P<a2> )?\n(?P<a3> \n)?OK( \(.*\))?$)"
+        # PyTest
+        r"|(^={10,} (((\d+ passed)?((, )?\d+ skipped)?(, \d+ warnings?)?)|(no tests ran)) in \d+\.\d+s ={10,}$)"
+    )
+
+    DEFAULT_FAILURE_REGEX = (
+        # Unittest
+        # The a{123} groups are to support Android logging differences.
+        # when https://github.com/chaquo/chaquopy/issues/746 is resolved,
+        # those groups can be deleted.
+        r"(^-{65,}\n(?P<a1> \n)?Ran \d+ tests in \d+\.\d{3}s\n(?P<a2> )?\n(?P<a3> \n)?FAILED( \(.*\))?$)"
+        # Pytest
+        r"|(^={10,} ((\d+ failed(, \d+ passed)?(, \d+ skipped)?)|(\d+ errors?)) in \d+.\d+s ={10,}$)"
```

Comment:
Ah - I forgot that you can generate an error with an exception in a fixture. However, I'm not sure I've ever seen warnings show up in the final summary - how are you manufacturing that? I've definitely seen warnings get *mentioned* in test output, but nothing I'm doing with my sample test suite is causing a warning count to appear in the summary

Reply:
It was a warning when using the existing Windows app stub:
```
============================== warnings summary ===============================
cygwin\git\beeware\toga\test\windows\app\Toga test\src\app_packages\_pytest\config\__init__.py:1172
  C:\Users\smith\cygwin\git\beeware\toga\test\windows\app\Toga test\src\app_packages\_pytest\config\__init__.py:1172: PytestConfigWarning: assertions not in tes
t modules or plugins will be ignored because assert statements are not executed by the underlying Python interpreter (are you using python -O?)

    self._warn_about_missing_assertion(mode)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
======================== 1 warning, 2 errors in 0.22s =========================
```